Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/jobs/yarn/App.java

```diff
@@ -0,0 +1,371 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * <p/>
+ * All rights reserved.
+ * <p/>
+ * MIT License
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.hdinsight.sdk.jobs.yarn;
+import com.fasterxml.jackson.annotation.JsonClassDescription;
+
+import java.util.List;
+
+/**
+ * Created by ltian on 5/6/2017.
+ */
+public class App {
```

Comment:
The fields types are missing, please refer the codes:

``` java
/**
 * An application resource contains information about a particular application that was submitted to a cluster.
 *
 * Based on Hadoop 2.7, refer to https://hadoop.apache.org/docs/stable/hadoop-yarn/hadoop-yarn-site/ResourceManagerRest.html#Cluster_Application_API
 *
 * Use the following URI to obtain an app object, from a application identified by the appid value.
 *   http://<rm http address:port>/ws/v1/cluster/apps/{appid}
 *
 * HTTP Operations Supported
 *   GET
 *
 * Query Parameters Supported
 *   None
 */
public class ClusterApplication {
    String id;                  // The application id
    String user;                // The user who started the application
    String name;                // The application name
    String applicationType;    // The application type
    String queue;               // The queue the application was submitted to
    String state;               // The application state according to the ResourceManager -
                                // valid values are members of the YarnApplicationState enum:
                                // NEW, NEW_SAVING, SUBMITTED, ACCEPTED, RUNNING, FINISHED, FAILED, KILLED
    String finalStatus;         // The final status of the application if finished -
                                // reported by the application itself - valid values are:
                                // UNDEFINED, SUCCEEDED, FAILED, KILLED
    float progress;             // The progress of the application as a percent
    String trackingUI;          // Where the tracking url is currently pointing -
                                // History (for history server) or ApplicationMaster
    String trackingUrl;         // The web URL that can be used to track the application
    String diagnostics;         // Detailed diagnostics information
    long clusterId;             // The cluster id
    long startedTime;           // The time in which application started (in ms since epoch)
    long finishedTime;          // The time in which the application finished (in ms since epoch)
    long elapsedTime;           // The elapsed time since the application started (in ms)
    String amContainerLogs;     // The URL of the application master container logs
    String amHostHttpAddress;   // The nodes http address of the application master
    int allocatedMB;            // The sum of memory in MB allocated to the application’s running containers
    int allocatedVCores;        // The sum of virtual cores allocated to the application’s running containers
    int runningContainers;      // The number of containers currently running for the application
    long memorySeconds;         // The amount of memory the application has allocated (megabyte-seconds)
    long vcoreSeconds;          // The amount of CPU resources the application has allocated (virtual core-seconds)

    public String getId() {
        return id;
    }

    public String getUser() {
        return user;
    }

    public String getName() {
        return name;
    }

    public String getApplicationType() {
        return applicationType;
    }

    public String getQueue() {
        return queue;
    }

    public String getState() {
        return state;
    }

    public String getFinalStatus() {
        return finalStatus;
    }

    public float getProgress() {
        return progress;
    }

    public String getTrackingUI() {
        return trackingUI;
    }

    public String getTrackingUrl() {
        return trackingUrl;
    }

    public String getDiagnostics() {
        return diagnostics;
    }

    public long getClusterId() {
        return clusterId;
    }

    public long getStartedTime() {
        return startedTime;
    }

    public long getFinishedTime() {
        return finishedTime;
    }

    public long getElapsedTime() {
        return elapsedTime;
    }

    public String getAmContainerLogs() {
        return amContainerLogs;
    }

    public String getAmHostHttpAddress() {
        return amHostHttpAddress;
    }

    public int getAllocatedMB() {
        return allocatedMB;
    }

    public int getAllocatedVCores() {
        return allocatedVCores;
    }

    public int getRunningContainers() {
        return runningContainers;
    }

    public long getMemorySeconds() {
        return memorySeconds;
    }

    public long getVcoreSeconds() {
        return vcoreSeconds;
    }

    public boolean isFinished() {
        String state = this.getState().toUpperCase();

        return state.equals("FINISHED") || state.equals("FAILED") || state.equals("KILLED");
    }
}
```